### PR TITLE
Data scoring

### PR DESCRIPTION
**Pull request by conig**
will do transformation separately.

### CODE REVIEW

File: R/cleaning/cardio_index.R [hunk 1]
```diff
@@ -33,11 +33,12 @@ get_cardio_index <- function(age,
   calc_z <- function(value, mean, sd) {
     (value - mean) / sd
   }
-
   inverted_HDL_C <- HDL_C * -1
+  age <- max(age, 6)
```
Comment (tarensanders) [inline R/cleaning/cardio_index.R]: Min age at the checkpoint wave was 11

File: R/score_data.R [hunk 2]
```diff
@@ -0,0 +1,32 @@
+score_data <- function(df_clean, bio_ref_data) {
+  df_clean$ApoBA1_ratio <- df_clean$apolipa1 / df_clean$apolipb
```
Comment (tarensanders) [inline R/score_data.R]: Wrong way round

File: R/score_data.R [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  # Diet classificaiton
+  df_clean$diet_SSB_flag <- df_clean$servessoftdrink != "I don't drink soft drinks, cordials or sports drinks"
+  # 2 or more serves of fruit
+  df_clean$diet_fruit_flag <- as.numeric(df_clean$servesfruit) >= 4
```
Comment (tarensanders) [inline R/score_data.R]: Should be 3 serves fruit from 2005 guidelines

File: R/score_data.R [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  # 5 or more serves of veg per day. Only 133 observations do!
+  df_clean$diet_veg_flag <- as.numeric(df_clean$servesveg) >= 7
```
Comment (tarensanders) [inline R/score_data.R]: Should be 4 serves veg

File: R/score_data.R [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  df_clean$diet <- ifelse(df_clean$diet_SSB_flag + df_clean$diet_fruit_flag + df_clean$diet_veg_flag == 3, "meets guidelines", "does not meet guidelines")
+
+  # Cardio score variable
+  df_clean$age_integer <- as.integer(round(df_clean$age,0))
+  df_clean$sex
```
Comment (tarensanders) [inline R/score_data.R]: ?

File: R/score_data.R [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  df_clean$cardio_index <- sapply(seq_len(nrow(df_clean)), function(i) {
+    get_cardio_index(
+      age = df_clean$age_integer[i],
+      sex = df_clean$sex[i],
+      SBP = df_clean$bpsys[i],
+      DBP = df_clean$bpdia[i],
+      waist_circumference = df_clean$waistcm[i],
+      total_body_fat = df_clean$bodyfat[i],
```
Comment (tarensanders) [inline R/score_data.R]: Do you want to delete these extra params since they don't contribute to cardio_index?

File: R/transform_data.R [hunk 7]
```diff
@@ -0,0 +1,10 @@
+transform_data <- function(scored_data) {
+  transformed_data <- scored_data |>
+    dplyr::mutate_all(as.character) |>
```
Comment (tarensanders) [inline R/transform_data.R]: Why as character?

**Review by tarensanders — CHANGES_REQUESTED** [review]

File: R/score_data.R (hunk 6 above)
Comment (conig) [reply R/score_data.R]: thanks, didn't catch that. The SBP and DBP are both needed though as we only have normed data individually (as far as im aware). Then I average that z score before averaging the z scores generally.

File: R/transform_data.R (hunk 7 above)
Comment (conig) [reply R/transform_data.R]: Another good catch. At first I was going long, then wide which required homogenous types. But as I'm skipping the long step that's no longer needed.

File: R/score_data.R (hunk 5 above)
Comment (conig) [reply R/score_data.R]: Have cleaned up the sex reference. That was mean screwing around checking variable names. need age integer for use in subsetting.

File: R/cleaning/cardio_index.R (hunk 1 above)
Comment (conig) [reply R/cleaning/cardio_index.R]: I'm applying for each observation where the minimum age is 4. is that not possible? 
image

**Review by tarensanders — APPROVED** [review]

**Pull request merged**